OpenOversight/app/main/forms.py

```diff
@@ -202,6 +202,9 @@ class AddOfficerForm(Form):
                             query_factory=unit_choices, get_label='descrip',
                             allow_blank=True, blank_text=u'None')
     employment_date = DateField('Employment Date', validators=[Optional()])
+    last_employment_date = DateField('Last Employment Date', validators=[Optional()])
+    last_employment_details = StringField('Last Employment Details', default='', validators=[
+        Regexp(r'\w*'), Length(max=50), Optional()])
```

Comment:
My first thought was, "shoot, you're right."  Then I tested this and was able to successfully enter last employment details with punctuation.

![OOGeneralInformation](https://user-images.githubusercontent.com/5529982/88495515-1b600180-cf88-11ea-8e5b-f8f7df1dbecd.png)

When I changed the validator from `Regexp(r'\w*')` to  `Regexp(r'\w')`, then I was no longer able to enter punctuation.  It appears that the wildcard adds flexibility to this regex.

Reply:
The issue here is that `Regexp(r'\w*')` doesn't do anything. It matches literally any string. `Regexp(r'\w')` only fails if the first character of the string is punctuation. So the question here really is which strings should be allowed and which shouldn't?

Reply:
I think my 2019 self used this regex because it's being used in most of the other string fields in `main/forms.py`.  Since @abandoned-prototype is right that it matches literally any string, I changed the regex to require alphanumeric characters in its input.  The field now allows punctuation, but will  throw an error if the input is only punctuation.